Analysis/Core/include/AnalysisCore/JetUtilities.h

```diff
@@ -0,0 +1,282 @@
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file JetUtilities.h
+/// \brief Jet related utilities
+///
+/// \author Raymond Ehlers <raymond.ehlers@cern.ch>, ORNL
+
+#ifndef O2_ANALYSIS_JETUTILITIES_H
+#define O2_ANALYSIS_JETUTILITIES_H
+
+#include <cmath>
+#include <limits>
+#include <numeric>
+#include <tuple>
+#include <vector>
+
+#include <TKDTree.h>
+
+#include "Framework/Logger.h"
+
+namespace JetUtilities
+{
+/**
+ * Duplicates jets around the phi boundary which are within the matching distance.
+ *
+ * NOTE: Assumes, but does not validate, that 0 <= phi < 2pi.
+ *
+ * @param jetsPhi Jets phi
+ * @param jetsEta Jets eta
+ * @param maxMatchingDistance Maximum matching distance. Only duplicate jets within this distance of the boundary.
+ */
+template <typename T>
+std::tuple<std::vector<std::size_t>, std::vector<T>, std::vector<T>> DuplicateJetsAroundPhiBoundary(
+  std::vector<T>& jetsPhi,
+  std::vector<T>& jetsEta,
+  double maxMatchingDistance,
+  // TODO: Remove additional margin after additional testing.
+  double additionalMargin = 0.05)
+{
+  const std::size_t nJets = jetsPhi.size();
+  std::vector<std::size_t> jetsMapToJetIndex(nJets);
+  // We need to keep track of the map from the duplicated vector to the existing jets.
+  // To start, we fill this map (practically, it maps from vector index to an index, so we
+  // just use a standard vector) with the existing jet indices, which range from 0..nJets.
+  std::iota(jetsMapToJetIndex.begin(), jetsMapToJetIndex.end(), 0);
+
+  // The full duplicated jets will be stored in a new vector to avoid disturbing the input data.
+  std::vector<T> jetsPhiComparison(jetsPhi);
+  std::vector<T> jetsEtaComparison(jetsEta);
+
+  // Duplicate the jets
+  // When a jet is outside of the desired phi range, we make a copy of the jet, shifting it by 2pi
+  // as necessary. The eta value is copied directly, as is the index of the original jet, so we can
+  // map from index in duplicated data -> index in original collection.
+  // NOTE: Assumes, but does not validate, that 0 <= phi < 2pi.
+  for (std::size_t i = 0; i < nJets; i++) {
+    // Handle lower edge
+    if (jetsPhi[i] <= (maxMatchingDistance + additionalMargin)) {
+      jetsPhiComparison.emplace_back(jetsPhi[i] + 2 * M_PI);
+      jetsEtaComparison.emplace_back(jetsEta[i]);
+      jetsMapToJetIndex.emplace_back(jetsMapToJetIndex[i]);
+    }
+    // Handle upper edge
+    if (jetsPhi[i] >= (2 * M_PI - (maxMatchingDistance + additionalMargin))) {
+      jetsPhiComparison.emplace_back(jetsPhi[i] - 2 * M_PI);
+      jetsEtaComparison.emplace_back(jetsEta[i]);
+      jetsMapToJetIndex.emplace_back(jetsMapToJetIndex[i]);
+    }
+  }
+
+  return std::move(std::make_tuple(jetsMapToJetIndex, jetsPhiComparison, jetsEtaComparison));
```

Comment:
Why not just `return { jetsMapToJetIndex, jetsPhiComparison, jetsEtaComparison };`?